Description: Don't rely on module state in teedataobject_clear
Origin: other, https://github.com/python/cpython/issues/115874#issuecomment-1965775536
Bug: https://github.com/python/cpython/issues/115874
Bug-Debian: https://bugs.debian.org/1063345

Index: b/Modules/itertoolsmodule.c
===================================================================
--- a/Modules/itertoolsmodule.c
+++ b/Modules/itertoolsmodule.c
@@ -832,8 +832,7 @@
         Py_CLEAR(tdo->values[i]);
     tmp = tdo->nextlink;
     tdo->nextlink = NULL;
-    itertools_state *state = get_module_state_by_cls(Py_TYPE(tdo));
-    teedataobject_safe_decref(tmp, state->teedataobject_type);
+    teedataobject_safe_decref(tmp, Py_TYPE(tdo));
     return 0;
 }
 
