# ntpath not in python-minimal

--- a/Lib/pathlib/_local.py
+++ b/Lib/pathlib/_local.py
@@ -1,5 +1,4 @@
 import io
-import ntpath
 import operator
 import os
 import posixpath
@@ -115,6 +114,7 @@ class PurePath(PurePathBase):
         return object.__new__(cls)
 
     def __init__(self, *args):
+        import ntpath
         paths = []
         for arg in args:
             if isinstance(arg, PurePath):
@@ -475,6 +475,7 @@ class PureWindowsPath(PurePath):
     On a Windows system, instantiating a PurePath should return this object.
     However, you can also instantiate it directly on any system.
     """
+    import ntpath
     parser = ntpath
     __slots__ = ()
 
