# ntpath not in python-minimal

--- a/Lib/pathlib/__init__.py
+++ b/Lib/pathlib/__init__.py
@@ -6,7 +6,6 @@ operating systems.
 """
 
 import io
-import ntpath
 import os
 import posixpath
 import sys
@@ -123,6 +122,7 @@ class PurePath(_abc.PurePathBase):
         return object.__new__(cls)
 
     def __init__(self, *args):
+        import ntpath
         paths = []
         for arg in args:
             if isinstance(arg, PurePath):
@@ -497,6 +497,7 @@ class PureWindowsPath(PurePath):
     On a Windows system, instantiating a PurePath should return this object.
     However, you can also instantiate it directly on any system.
     """
+    import ntpath
     parser = ntpath
     __slots__ = ()
 
